File: Atcoder_#301/ruby/c_atCoder_cards.rb
def can_win?(s, t)
  # @ に変換可能な文字を定義
  atcoder_chars = ['a', 't', 'c', 'o', 'd', 'e', 'r']

  # 各文字のカウントを行う
  s_count = Hash.new(0)
  t_count = Hash.new(0)

  s.chars.each { |ch| s_count[ch] += 1 unless ch == '@' }
  t.chars.each { |ch| t_count[ch] += 1 unless ch == '@' }

  # S と T の @ の数を取得
  s_at = s.count('@')
  t_at = t.count('@')

  # S側とT側の文字を比較し、足りない分を@で補えるか確認
  atcoder_chars.each do |ch|
    if s_count[ch] > t_count[ch]
      t_at -= (s_count[ch] - t_count[ch])
    elsif t_count[ch] > s_count[ch]
      s_at -= (t_count[ch] - s_count[ch])
    end
    # @が足りなくなったら不可能
    return false if s_at < 0 || t_at < 0
  end

  # @ を使わない部分の文字もチェック
  (('a'..'z').to_a - atcoder_chars).each do |ch|
    return false if s_count[ch] != t_count[ch]
  end

  true
end

# 入力の取得
s = gets.chomp
t = gets.chomp

# 結果を判定して出力
if can_win?(s, t)
  puts 'Yes'
else
  puts 'No'
end

File: Atcoder_#301/javascript/c_atCoder_cards.js
function canWin(s, t) {
    // @ に変換可能な文字を定義
    const atcoderChars = new Set(['a', 't', 'c', 'o', 'd', 'e', 'r']);

    // 各文字のカウントを行う
    const sCount = {};
    const tCount = {};

    // カウント初期化関数
    function countChars(str, countObj) {
        for (let ch of str) {
            if (ch !== '@') {
                countObj[ch] = (countObj[ch] || 0) + 1;
            }
        }
    }

    countChars(s, sCount);
    countChars(t, tCount);

    // S と T の @ の数を取得
    let sAt = (s.match(/@/g) || []).length;
    let tAt = (t.match(/@/g) || []).length;

    // S側とT側の文字を比較し、足りない分を@で補えるか確認
    for (let ch of atcoderChars) {
        const sCountCh = sCount[ch] || 0;
        const tCountCh = tCount[ch] || 0;

        if (sCountCh > tCountCh) {
            tAt -= sCountCh - tCountCh;
        } else if (tCountCh > sCountCh) {
            sAt -= tCountCh - sCountCh;
        }

        // @が足りなくなったら不可能
        if (sAt < 0 || tAt < 0) {
            return false;
        }
    }

    // @ を使わない部分の文字もチェック
    for (let ch in sCount) {
        if (!atcoderChars.has(ch) && sCount[ch] !== (tCount[ch] || 0)) {
            return false;
        }
    }

    for (let ch in tCount) {
        if (!atcoderChars.has(ch) && tCount[ch] !== (sCount[ch] || 0)) {
            return false;
        }
    }

    return true;
}

// 標準入力の取得と結果の出力
function main(input) {
    const lines = input.trim().split('\n');
    const s = lines[0];
    const t = lines[1];

    if (canWin(s, t)) {
        console.log('Yes');
    } else {
        console.log('No');
    }
}

// 標準入力を読み込み、main関数を呼び出し
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin', 'utf8');
main(input);
